app/ide-desktop/lib/client/src/config/parser.ts

```diff
@@ -0,0 +1,319 @@
+
+    display(): string {
+        const value = this.value == null ? '' : `=${this.value}`
+        return `--${this.name}${value}`
+    }
+}
+
+/** Replaces `-no-...` with `--no-...`. This is a hotfix for Yargs bug:
+ * https://github.com/yargs/yargs-parser/issues/468 */
+function fixArgvNoPrefix(argv: string[]): string[] {
+    const singleDashPrefix = '-no-'
+    const doubleDashPrefix = '--no-'
+    return argv.map(arg => {
+        if (arg.startsWith(singleDashPrefix)) {
+            return doubleDashPrefix + arg.slice(singleDashPrefix.length)
+        } else {
+            return arg
+        }
+    })
+}
+
+function argvAndChromeOptions(processArgs: string[]): {
+    argv: string[]
+    chromeOptions: ChromeOption[]
+} {
+    const chromeOptionRegex = /--?chrome.([^=]*)(=(.*))?/
+    const argv = []
+    const chromeOptions: ChromeOption[] = []
+    for (let i = 0; i < processArgs.length; i++) {
+        const processArg = processArgs[i]
+        if (processArg != null) {
+            const match = processArg.match(chromeOptionRegex)
+            if (match != null) {
+                const optionName = match[1] as string
+                const optionValue = match[3]
+                if (optionValue != null) {
+                    chromeOptions.push(new ChromeOption(optionName, optionValue))
+                } else {
+                    const nextArgValue = processArgs[i + 1]
+                    if (nextArgValue != null && !nextArgValue.startsWith('-')) {
+                        chromeOptions.push(new ChromeOption(optionName, nextArgValue))
+                        i++
+                    } else {
+                        chromeOptions.push(new ChromeOption(optionName, undefined))
+                    }
+                }
+            } else {
+                argv.push(processArg)
+            }
+        }
+    }
+    return { argv, chromeOptions }
+}
+
+// =====================
+// === Option Parser ===
+// =====================
+
+export function parseArgs() {
+    const args = config.config
+    const { argv, chromeOptions } = argvAndChromeOptions(fixArgvNoPrefix(hideBin(process.argv)))
+
+    const yargOptions = args.optionsRecursive().reduce((opts: { [key: string]: any }, option) => {
+        const yargsParam = Object.assign({}, option)
+        // @ts-ignore
+        yargsParam.requiresArg = ['string', 'array'].includes(yargsParam.type)
+        // @ts-ignore
+        yargsParam.default = undefined
+        // @ts-ignore
+        opts[naming.camelToKebabCase(option.qualifiedName())] = yargsParam
+        return opts
+    }, {})
+
+    let optParser = yargs()
+        .version(false)
+        .parserConfiguration({
+            // Allow single-dash arguments, like `-help`.
+            'short-option-groups': false,
+            // Treat dot-arguments as string keys, like `foo.bar`.
+            'dot-notation': false,
+            // Makes all flags passed after '--' be one string.
+            'populate--': true,
```

Comment:
Why do we want that? I'm afraid that there are some escaping situations that could break on some shells.

Reply:
We were always using that, this is not a change. However, to answer specifically, otherwise we are just not handling this situation and arguments after `--` are just ignored.